spec/ffg.tla

```diff
@@ -281,7 +281,7 @@ Chain(x, node_state, N) ==
                 IF P(seq[1])
                 THEN seq
                 ELSE <<b(seq[1])>> \o seq \* Alternatively, we can append here and reverse the list at the end
-        IN ApaFoldSeqLeft( step, <<x>>, MkSeq(N, LAMBDA i: i) )
+        IN ApaFoldSet( step, <<x>>, 1..N )
```

Comment:
In principle, we could filter here to `{ i \in 1..N : i <= (CHOOSE initial_target \in DOMAIN x : TRUE).chkp_slot }`.

However, we should anyhow introduce an optimization that invokes `Chain` with a map for all relevant checkpoints, to avoid recomputing the fold.

Reply:
Have you compared the runtimes of these 2 empirically? @konnov and I had some discussions around this, and the reason we opted not to use set-fold, is because it always has to add the constraints ensuring set element uniqueness (despite knowing that for the range 1..N, the overapproximation is tight).
One of the solutions we were floating is to introduce another kind of operation to Apalache, which is basically a
N-tuple iteration of an operator, that wouldn't rely on an underlying set- or sequence-representation of 1 to N.

Reply:
Oh that's a good point!

Let me compare it real quick, but given that these folds are the most expensive part of the protocol, it probably makes sense to rewrite all of them as `MkSeq`.

Reply:
Not a significant difference, but still makes sense to use `FoldSeq` / `MkSeq` everywhere. PTAL